core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -526,6 +533,21 @@ private static AnnotatedValueResolver ofQueryParam(String name,
                 .build();
     }
 
+    private static AnnotatedValueResolver ofFileParam(String name,
+                                                      AnnotatedElement annotatedElement,
+                                                      AnnotatedElement typeElement, Class<?> type,
+                                                      @Nullable String description) {
+        return new Builder(annotatedElement, type)
+                .annotationType(Param.class)
+                .httpElementName(name)
+                .typeElement(typeElement)
+                .supportContainer(true)
+                .description(description)
+                .aggregation(AggregationStrategy.ALWAYS)
```

Comment:
Shouldn't it be `AggregationStrategy.NONE` because it is not aggregated with? `req.aggregate()`?

Reply:
If using NONE we can't distinguish lazy `Multipart` or eager `@Param File` at `#aggregationType(AggregationStrategy strategy, RequestHeaders headers)`. 

And there is maybe
`public HttpResponse foo(@Param String name)` using multipart to send. So for this we may still treat FOR_FORM_DATA become `AggregationType.MULTIPART`.

Reply:
I treat aggregate includes saving to file.
Not sure if it can be applied to `exchangeType` also.

Reply:
I see. We may need to refactor `AggregationStrategy` later to fluently express the current requirements of the request aggregation. 
Could we apply the following change?
```diff
@@ -489,7 +490,7 @@ public final class AnnotatedService implements HttpService {
     @Override
     public ExchangeType exchangeType(RequestHeaders headers, Route route) {
         // TODO(ikhoon): Support a non-streaming response type.
-        if (AggregationStrategy.aggregationRequired(aggregationStrategy, headers)) {
+        if (AnnotatedValueResolver.aggregationType(aggregationStrategy, headers) == AggregationType.ALL) {

```

Reply:
Nice catch. Wanna double confirm that MULTIPART needs to be `BIDI_STREAMING`, right? Because if we use `RESPONSE_STREAMING`, it may buffer somewhere?